cura/CuraPackageManager.py

```diff
@@ -47,3 +63,30 @@ def getMachinesUsingPackage(self, package_id: str) -> Tuple[List[Tuple[GlobalSta
                         machine_with_qualities.append((global_stack, str(extruder_nr), container_id))
 
         return machine_with_materials, machine_with_qualities
+
+    def getAllLocalPackages(self) -> List[Dict[str, Any]]:
+        """ returns an unordered list of all the package_info installed, to be installed or to be returned"""
+
+        class PkgInfo:
+            # Needed helper class because a dict isn't hashable
```

Comment:
Dict can also be hashed if you do this: 

class hashabledict(dict):
    def __hash__(self):
        return hash(tuple(sorted(self.items())))

Reply:
But then the hash is calculated over the complete dict and I'm just interested in the `package_id`